enhancements/update/configurable-clusterversion-signature-stores.md

```diff
@@ -0,0 +1,188 @@
+---
+title: configurable-clusterversion-signature-stores
+authors:
+  - "@PratikMahajan"
+reviewers:
+  - "@LalatenduMohanty"
+  - "@wking"
+approvers:
+  - "@LalatenduMohanty"
+api-approvers:
+  - "@deads2k"
+creation-date: 2023-09-27
+last-updated: 2023-10-09
+tracking-link:
+  - https://issues.redhat.com/browse/OTA-916
+---
+
+# ClusterVersion has option to replace default signature stores
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] Operational readiness criteria is defined
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+CVO currently fetches the upstream urls for signature verification through [cluster-update-keys]. 
+To increase flexibility and improve disconnected cluster user experience, we need to provide a way for the CVO to 
+get the required signatures from custom signature stores or the OSUS running in the disconnected environment.
+
+
+## Motivation
+
+Today, in the disconnected environments, a user has to clone the signatures config map as well as the release images 
+then apply the configMap to each cluster for the update to proceed successfully.
+With [serve image signatures for disconnected environments], OSUS will be able to serve the cloned image signatures locally.
+
+Currently, there is no way for the CVO to fetch the signatures through custom URIs. This enhancement defines an API change in the CVO
+to accept a custom update-service URI which can be used to verify release signatures. 
+
+
+### User Stories
+
+* As a cluster administrator, I want the cluster's CVO to add new signature store URIs.
+* As a cluster administrator, I want the cluster's CVO to replace the existing signature store URIs with a user provided URI
+
+### Goals
+
+* To verify the release image signature in disconnected environments using OSUS signatures URI.
+* To replace existing signature stores with custom user-provided stores.
+
+### Non-Goals
+
+* This enhancement does not cover how OSUS running in the disconnected environment gets the signatures and if the 
+signatures are validated during the mirroring process.
+
+## Proposal
+
+### Workflow Description
+
+1. Cluster admin will have to add the verification URI to CVO
+2. CVO will reach the newly added URI to verify the release images.
+
+### API Extensions
+
+* CVO will grow a new field, an array, which will be populated with the image verification URI.
+* This will be an array as it'll allow the admin to use multiple locally configured stores. 
+* Existing default signature stores will be used as fallback in-case custom stores don't contain signatures or are offline. 
```

Comment:
So will the default stores always be used then? Whether this field is configured or not?